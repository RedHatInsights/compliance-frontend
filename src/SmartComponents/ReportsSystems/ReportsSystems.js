import React from 'react';
import SystemsTable from '../SystemsTable/SystemsTable';
import { ReportTabs } from 'PresentationalComponents';
import { PageHeader, PageHeaderTitle, Main } from '@redhat-cloud-services/frontend-components';

const ReportsSystems = () => {
    const columns = [{
        key: 'facts.compliance.display_name',
        title: 'Name',
        props: {
            width: 40
        }
    }, {
        key: 'facts.compliance.policies',
        title: 'Policies',
        props: {
            width: 40
        }
    }, {
        key: 'facts.compliance.compliance_score',
        title: 'Compliance score',
        props: {
            width: 10
        }
    }, {
        key: 'facts.compliance.last_scanned',
        title: 'Scan date',
        props: {
            width: 10
        }
    }];

    return (
        <React.Fragment>
            <PageHeader className='page-header'>
                <PageHeaderTitle title="Compliance reports" />
                <ReportTabs/>
            </PageHeader>
            <Main>
                <SystemsTable
                    compliantFilter
                    showOnlySystemsWithTestResults
                    columns={columns} />
            </Main>
        </React.Fragment>
    );
};

export default ReportsSystems;
