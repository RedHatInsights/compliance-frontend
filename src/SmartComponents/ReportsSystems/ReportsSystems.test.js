import toJson from 'enzyme-to-json';

import ReportsSystems from './ReportsSystems.js';

jest.mock('@redhat-cloud-services/frontend-components-inventory-compliance', () =>
    () => 'Mocked ComplianceSystemDetails'
);

describe('ReportsSystems', () => {
    it('expect to render without error', () => {
        const wrapper = shallow(
            <ReportsSystems />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
