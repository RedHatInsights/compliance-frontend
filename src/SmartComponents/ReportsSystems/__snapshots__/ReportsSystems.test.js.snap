// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportsSystems expect to render without error in beta 1`] = `
<Fragment>
  <PageHeader
    className="page-header-tabs"
  >
    <PageHeaderTitle
      title="Compliance reports"
    />
    <withRouter(Connect(RouterParams)) />
  </PageHeader>
  <Connect(Main)>
    <Connect(withApollo(SystemsTable))
      columns={
        Array [
          Object {
            "key": "facts.compliance.display_name",
            "props": Object {
              "width": 40,
            },
            "title": "Name",
          },
          Object {
            "key": "facts.compliance.policies",
            "props": Object {
              "width": 40,
            },
            "title": "Policies",
          },
          Object {
            "key": "facts.compliance.compliance_score",
            "props": Object {
              "width": 10,
            },
            "title": "Compliance score",
          },
          Object {
            "key": "facts.compliance.last_scanned",
            "props": Object {
              "width": 10,
            },
            "title": "Scan date",
          },
        ]
      }
      compliantFilter={true}
      showOnlySystemsWithTestResults={true}
    />
  </Connect(Main)>
</Fragment>
`;

exports[`ReportsSystems expect to render without error in stable 1`] = `
<Fragment>
  <PageHeader
    className="page-header-tabs"
  >
    <PageHeaderTitle
      title="Compliance reports"
    />
    <withRouter(Connect(RouterParams)) />
  </PageHeader>
  <Connect(Main)>
    <Connect(withApollo(SystemsTable))
      columns={
        Array [
          Object {
            "key": "facts.compliance.display_name",
            "props": Object {
              "width": 40,
            },
            "title": "Name",
          },
          Object {
            "key": "facts.compliance.policies",
            "props": Object {
              "width": 40,
            },
            "title": "Policies",
          },
          Object {
            "key": "facts.compliance.compliance_score",
            "props": Object {
              "width": 10,
            },
            "title": "Compliance score",
          },
          Object {
            "key": "facts.compliance.last_scanned",
            "props": Object {
              "width": 10,
            },
            "title": "Scan date",
          },
        ]
      }
      compliantFilter={true}
      showOnlySystemsWithTestResults={true}
    />
  </Connect(Main)>
</Fragment>
`;
